src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -3206,4 +3206,46 @@ public static void createModuleInstsFromModuleImplsInsts(Design design, Collecti
         }
     }
 
+    /**
+     * Determine if a Net is driven by a hierarchical port, created as part of an out-of-context
+     * synthesis flow, for example.
+     * @param net Net to examine.
+     * @returns True if driven by a hierport.
+     */
+    public static boolean isNetDrivenByHierPort(Net net) {
+        if (net.getSource() != null) {
+            // Net can only be driven by a hier port if it has no site pin driver
+            return false;
+        }
+
+        if (net.isStaticNet()) {
+            // Static nets cannot be driven by a hier port
+            return false;
+        }
+
+        EDIFNet en = net.getLogicalNet();
+        List<EDIFPortInst> sourcePorts = en.getSourcePortInsts(true);
+        if (sourcePorts.isEmpty()) {
+            // Net has no source ports; truly an driver-less net
+            return false;
+        }
+
+        if (sourcePorts.size() != 1) {
+            return false;
+        }
+
+        EDIFPortInst epi = sourcePorts.get(0);
+        if (!epi.isTopLevelPort()) {
+            // Hier ports must be top level ports
+            return false;
+        }
+
+        Cell cell = net.getDesign().getCell(epi.getName());
```

Comment:
@clavin-xlnx Is it correct to assume that the corresponding `Cell` of a top-level `EDIFPortInst` would be the named the same?

Reply:
In general, that tends to be the behavior of designs generated by Vivado.  However, I don't think it is strictly enforced or required.

Reply:
Would you recommend removing this check, or is there some other way of going from top-level `EDIFPortInst` to `Cell`? It is an overly-cautious check since you would expect `Net.getSource()` to not be `null` if there was a I/O `Cell`...

Reply:
I'm ok to remove it.  At this point in the code, `epi` is a top level port and there would be no way to get the cell from it as it references its parent instance, not a cell instance in the context of the cell.

Reply:
Done in de02d9e!